history.go

```diff
@@ -54,11 +68,15 @@ func (h *History) UpdateHistory(policies Policies, m *MerkleTree, nextEp Epoch)
 		}
 		h.loadedEpochs = append(h.loadedEpochs[:0], h.loadedEpochs[n:]...)
 	}
-	h.snapshots[nextEp] = nextStr
-	h.loadedEpochs = append(h.loadedEpochs, nextEp)
+
+	h.updateInternal(policies, nextEp)
 	return nil
 }
 
+func (h *History) Set(key string, value []byte) error {
```

Comment:
I also just realized that a directory lookup also needs to return an STR, so I think this is another reason why we should move these wrappers into the server.